File: practice-webmvc/src/main/java/com/jiwoon/practicewebmvc/member/MemoryMemberRepository.java
package com.jiwoon.practicewebmvc.member;

import org.springframework.stereotype.Component;

import java.util.Map;
import java.util.concurrent.ConcurrentHashMap;

@Component
public class MemoryMemberRepository implements MemberRepository {

    private static Map<Long, Member> store = new ConcurrentHashMap<>();

    @Override
    public Member findById(long memberId) {
        return store.get(memberId);
    }

    @Override
    public void save(Member member) {
        store.put(member.getMemberId(), member);
    }
}

File: spring-webflux-kotlin/src/main/kotlin/com/bcp0109/springwebfluxkotlin/domain/MemoryMemberRepository.kt
package com.bcp0109.springwebfluxkotlin.domain

import org.springframework.stereotype.Repository
import reactor.core.publisher.Mono
import java.util.concurrent.ConcurrentHashMap

@Repository
class MemoryMemberRepository : MemberRepository {

    companion object {
        val store = ConcurrentHashMap<Long, Member>()
    }

    override fun save(member: Member): Mono<Member> {
        member.id = store.size + 1L
        store[member.id!!] = member
        return Mono.just(member)
    }

    override fun findById(id: Long): Mono<Member> {
        return Mono.justOrEmpty(store[id])
    }

}
